src/wui/interactive_gamebase.cc

```diff
@@ -299,16 +311,54 @@ void InteractiveGameBase::gamespeed_menu_selected(GameSpeedEntry entry) {
 	}
 }
 
-void InteractiveGameBase::increase_gamespeed() {
+void InteractiveGameBase::increase_gamespeed(uint32_t speed) {
 	if (GameController* const ctrl = get_game()->game_controller()) {
-		ctrl->set_desired_speed(ctrl->desired_speed() + 1000);
+		uint32_t const current_speed = ctrl->desired_speed();
+		switch (speed) {
+		case 250:
+			ctrl->set_desired_speed(current_speed + speed);
+			break;
+		case 1000:
+			if (current_speed < 1000) {
+				ctrl->set_desired_speed(1000);
+			} else {
+				ctrl->set_desired_speed(current_speed - current_speed % speed + speed);
+			}
+			break;
+		case 10000:
+			ctrl->set_desired_speed(
+			   current_speed >= speed ? current_speed - current_speed % speed + speed : speed);
+			break;
+		default:
+			break;
```

Comment:
Can this happen? Just increase the desired speed by the requested value and add a log() call for the unexpected argument.

Reply:
And use the `constexpr`s I requested in a another comment above as case labels here